### ISSUE TITLE
Out of memory

### ISSUE THREAD

**Issue by rebelhn** [cupslab/neural_network_cracking#30]
I've GeForce GT620, Cuda 8.0 with Keras and Theano installed.
When run password guessing with input 1384 password, there are ong warning and two error orcurred:
- Warning: Using gpu device 0: GeForce GT 620 (CNMeM is disabled, cuDNN not available)
/home/rebel/Theano-rel-0.8.2/theano/tensor/signal/downsample.py:6: UserWarning: downsample module has been moved to the theano.tensor.signal.pool module.
  "downsample module has been moved to the theano.tensor.signal.pool module.")
I've changed to use pool() but it has error pool() is not take position at "ds".
- Error: 
Error allocating 400000000 bytes of device memory (out of memory). Driver report 99483648 bytes free and 1011351552 bytes total 
Uncaught exception!
 Error allocating 400000000 bytes of device memory (out of memory).
Apply node that caused the error: GpuDot22(GpuReshape{2}.0, GpuReshape{2}.0)
Toposort index: 115
Inputs types: [CudaNdarrayType(float32, matrix), CudaNdarrayType(float32, matrix)]
Inputs shapes: [(100000, 51), (51, 1000)]
Inputs strides: [(51, 1), (1000, 1)]
Inputs values: ['not shown', 'not shown']
Outputs clients: [[GpuReshape{3}(GpuDot22.0, MakeVector{dtype='int64'}.0)]]
There are my input and config files:
[alypaa.txt](https://github.com/cupslab/neural_network_cracking/files/4614316/alypaa.txt)
[guess_len8_config.json.txt](https://github.com/cupslab/neural_network_cracking/files/4614326/guess_len8_config.json.txt)

**Comment by m33x** [comment]
There is no fix for this issue. Your GPU (graphics card) is simply too old (it’s from 2012), and has too little video memory. The card is officially out-of-support by Nvidia. Getting a more recent model like a

RTX2070 - 8GB - 430€
RTX2060 - 6GB - 340€
GTX1660 SUPER - 6GB - 250€
GTX1650 SUPER - 4GB - 180€

is highly recommended.

**Comment by rebelhn** [comment]
Yes i know this card is too old. Just do a testing with this project, if success i will buy some new. Can i resolve this by use more than one GT620, maybe 2 or 3 cards?

**Comment by rebelhn** [comment]
Does this project only run on first GPU card? How can i make it run on multiple cards?
